jmxfetch.py

```diff
@@ -274,26 +276,27 @@ def _start(self, path_to_java, java_run_opts, jmx_checks, command, reporter, too
 
             log.info("Running %s" % " ".join(subprocess_args))
 
-            # Launch JMXfetch subprocess
-            jmx_process = subprocess.Popen(
-                subprocess_args,
-                close_fds=not redirect_std_streams,  # set to True instead of False when the streams are redirected for WIN compatibility
-                stdout=subprocess.PIPE if redirect_std_streams else None,
-                stderr=subprocess.PIPE if redirect_std_streams else None
-            )
-            self.jmx_process = jmx_process
-
-            # Register SIGINT and SIGTERM signal handlers
-            self.register_signal_handlers()
-
-            if redirect_std_streams:
-                # Wait for JMXFetch to return, and write out the stdout and stderr of JMXFetch to sys.stdout and sys.stderr
-                out, err = jmx_process.communicate()
-                sys.stdout.write(out)
-                sys.stderr.write(err)
-            else:
-                # Wait for JMXFetch to return
-                jmx_process.wait()
+            # Launch JMXfetch subprocess manually, w/o get_subprocess_output(), since it's a special case
+            with nested(tempfile.TemporaryFile('rw'), tempfile.TemporaryFile('rw')) as (stdout_f, stderr_f):
+                jmx_process = subprocess.Popen(
+                    subprocess_args,
+                    close_fds=not redirect_std_streams,  # set to True instead of False when the streams are redirected for WIN compatibility
```

Comment:
Comes from one of my commits actually but the comment here is wrong, it's not `set to True when the streams are redirected` but rather `only set to True when the streams are not redirected`. Could you correct it please? :)